Provider/provider_attribute_type.md

```diff
@@ -0,0 +1,117 @@
+    *preferredHandlerClassname* | `Handler` | Handler sub resource for the Custom Data Type used.Can optionally define a specific handler class want to use (otherwise the framework will choose the best handler for the chosen datatype).To find which handlers to use for the Custom DataType please refer here
+    *datatypeConfig* | `String` | Allow the data type have any name and config it wants/needs.
+    *handlerConfig* | `String` | Allow the handler have any name and config it wants/needs.This will help to identify the data type unambiguously which has been contained and will allow introspecting
+
+    ```console
+        POST /providerattributetype
+        {
+          "name": "Provider Location",
+          "description": "This attribute type will record the loication of the provider",
+          "datatypeClassname": "org.openmrs.customdatatype.datatype.LongFreeTextDatatype",
+          "minOccurs": 0,
+          "maxOccurs": 1,
+          "datatypeConfig": "default",
+          "preferredHandlerClassname": "org.openmrs.web.attribute.handler.LongFreeTextTextareaHandler",
+          "handlerConfig": "dafault"
+        }
+    ```
+### Update a provider attribute type
+
+*  Update a target provider attribute type with given UUID, this method only modifies properties in the request. Returns a `404 Not Found`
+status if the provider attribute not exists. If the user not logged in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Attributes
+
+      Parameter | Type | Description
+      --- | --- | ---
+      *name* | `String` | Name of the provider attribute type
+      *description* | `String` | Description (Required)
+      *datatypeClassname* | `CustomDataType Resource` | Data type for the attribute type resource.OpenMRS provides **Custom data type resource** which gives flexibility to select the data type accordingly (Required)
+      *minOccurs* | `Number` | Minimum number of times this value can be specified for a single provider.Use `0` or `1` as the default value (Required) (Required)
+      *maxOccurs* | `Number` | Maximum number of times this value can be specified for a single provider (e.g., use 1 to prevent an attribute from being added to a provider multiple times)
+      *preferredHandlerClassname* | `Handler` |  Handler sub resource for the Custom Data Type used.Can optionally define a specific handler class want to use (otherwise the framework will choose the best handler for the chosen datatype).To find which handlers to use for the Custom DataType please refer here
+      *datatypeConfig* | `String` | Allow the data type have any name and config it wants/needs.
+      *handlerConfig* | `String` | Allow the handler have any name and config it wants/needs.This will help to identify the data type unambiguously which has been contained and will allow introspecting
```

Comment:
Description is confusing. I believe the `handlerConfig` is sent directly to the handler, allowing a handler to be used for more than one attribute type. The actual configuration depends on the needs of the specified handler. For example, a "Pre-defined List" handler could be made to implement a simple selection list and this configuration would tell the handler the possible choices in the list for this specific attribute type.

Also need space after period.